api/src/main/java/org/apache/iceberg/expressions/Unbound.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.expressions;
+
+import org.apache.iceberg.types.Types;
+
+public interface Unbound<T, B> {
```

Comment:
Does it have to extend `ValueExpression`?

Reply:
Well, it looks like it doesn't. I was confused as `Bound` is `ValueExpression` but `Unbound` isn't.

Reply:
Not all expressions are bound or unbound. Only those with references are. For example, `true` is neither bound nor unbound. That's why these are separate properties of an expression.

Reply:
Updated so `Bound` and `Unbound` are independent of `ValueExpression` (now called `Term`).